src/lib/cache/gdfs_cache.hpp

```diff
@@ -35,7 +35,7 @@ class GDFSCache : public AbstractCache<Key, Value> {
 
   explicit GDFSCache(size_t capacity = DEFAULT_CACHE_CAPACITY) : AbstractCache<Key, Value>(capacity), _inflation(0.0) {}
 
-  void set(const Key& key, const Value& value, double cost = 1.0, double size = 1.0) {
+  void set(const Key& key, const Value& value, double cost = 1.0, double size = 1.0) final {
```

Comment:
No real pattern in whether I use override or final. I just went with what was predominant in the files.

#1254

Reply:
It does. Non-virtual functions are final by definition. `virtual final` does not make sense as you would declare your method virtual on the top-level, but not allow anyone to override it. Thus, whenever you mark something as final, it is clear that you are overriding something.

I was trying to find a source for this, and found this, which seems to be wrong: https://stackoverflow.com/questions/29412412/does-final-imply-override

Clang complains about final being used on something that does not override a virtual method:

https://godbolt.org/z/MarY1G

Reply:
Learned something. Thanks!